static/src/javascripts/projects/common/modules/commercial/commercial-feature-policies.js

```diff
@@ -68,16 +68,15 @@ define([
         }
     };
 
-    policies.membershipMessages = function () {
-        if (!detect.adblockInUseSync() &&
-            detect.getBreakpoint() !== 'mobile' &&
-            config.page.contentType === 'Article' &&
-            !userFeatures.isPayingMember()
-        ) {
-            return {
-                membershipMessages : true
-            };
-        }
+    policies.checkWeCanShowMembershipMessages = function () {
```

Comment:
> i'd rather see this I think: {code}

I was thinking of refactoring the module on similar lines. The notion of 'policies' which mapped 'concepts' to 'feature diffs' sounded interesting on paper, but has caused confusion in practice, and most changes to this file have involved creating policies for the case of turning on / off individual flags, which is quite boilerplatey. The idea just hasn't worked well.

>  shall i change it back?

No, I agree the invocation with `then` is more readable, and like that the name suggests a different interface. I think it's better than reverting to the previous name.